src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocketOptions.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+
+namespace System.Net.WebSockets
+{
+    public sealed class ClientWebSocketOptions
+    {
+        private bool _isReadOnly; // After ConnectAsync is called the options cannot be modified.
+        private readonly List<string> _requestedSubProtocols;
+        private readonly WebHeaderCollection _requestHeaders;
+        private TimeSpan _keepAliveInterval;
+        private ICredentials _credentials;
+        private IWebProxy _proxy;
+        private X509CertificateCollection _clientCertificates;
+        private CookieContainer _cookies;
+
+        internal ClientWebSocketOptions()
+        {
+            _requestedSubProtocols = new List<string>();
+            _requestHeaders = new WebHeaderCollection();
```

Comment:
@justinvp /cc @SidharthNabar ClientWebSocketOptions will not have effect within this initial implementation. We're tracking a proper implementation.